cpp/src/arrow/filesystem/gcsfs_test.cc

```diff
@@ -591,6 +634,7 @@ TEST_F(GcsIntegrationTest, DeleteDirSuccess) {
   arrow::fs::AssertFileInfo(fs.get(), PreexistingBucketPath(), FileType::Directory);
   arrow::fs::AssertFileInfo(fs.get(), PreexistingObjectPath(), FileType::File);
   for (auto const& info : hierarchy.contents) {
+    if (!fs::internal::IsAncestorOf(hierarchy.base_dir, info.path())) continue;
```

Comment:
You could actually check that this entry still exists instead of ignoring it.
(same comment below too)

Reply:
Done.